src/__tests__/datasource.test.js

```diff
@@ -135,4 +132,18 @@ describe('Mongoose', () => {
     expect(res1[0].name).toBe('Bob')
     expect(res2[0]).toBeUndefined()
   })
+
+  test('Data Source with Context', async () => {
+    const users = new Users({ modelOrCollection: UserModel, context: { token: '123' }})
+    
+    expect(users.context.token).toBe('123')
+  })
+
+  test('Data Source with Context that contains a User', async () => {
+    const users = new Users({ modelOrCollection: userCollection, context: { user: alice }})
+    const user = await users.findOneById(alice._id)
+
+    expect(user.name).toBe('Alice')
+    expect(users.context.user.name).toBe(user.name)
+  })
```

Comment:
@lorensr I added these two tests to check the behavior for providing Context to datasources in Apollo Server 4. I also made minor changes to the construction of the Users datasource. Let me know if you think more tests are necessary. I also added the old README file and added a note to both which explains the versioning.